numpy/core/setup_common.py

```diff
@@ -216,6 +216,24 @@ def check_long_double_representation(cmd):
         except (AttributeError, ValueError):
             pass
 
+    # Disable multi-file interprocedural optimization in the Intel compiler on Linux
+    # which generates intermediary object files and prevents checking the
+    # float representation.
+    elif sys.platform != "win32" and cmd.compiler.compiler_type.startswith('intel') \
+            and '-ipo' in cmd.compiler.cc_exe:
+        try:
+            newcompiler = cmd.compiler.cc_exe.replace(' -ipo', '')
+            cmd.compiler.set_executables(
+                compiler=newcompiler,
+                compiler_so=newcompiler,
+                compiler_cxx=newcompiler,
+                linker_exe=newcompiler,
+                linker_so=newcompiler + ' -shared'
+            )
+        except (AttributeError, ValueError):
```

Comment:
Is there a reason to expect these errors?

Reply:
Yes, in every case I've seen including my own builds, building numpy with the `-ipo` flag for `icc` invariably makes it fail that test.  After adding this fix, I was able to build successfully and pass all of the builtin tests. multi-file ipo has potential to improve performance significantly, as it allows inlining, dead call optimization, and so forth between multiple files and is useful for HPC applications.

Reply:
I'm not entirely sure. I'm far from being a python master and rarely use it, except for when I'm supporting scientists at work, so I'm not 100% familiar with the exceptions that can be thrown from certain methods.  I included that to be consistent with the previous check for MSVC that is doing something very similar.